Doc 1776/open banking partner documentation on docsswanio (#641)
swan-io/docs: docs/topics/open-banking/ais.mdx

    TPP-->>SoftwareApp: Delivers data, which the software displays to the user
```

## Common AIS use cases {#use-cases}

- Accounting and financial management software importing bank transactions automatically.
- Personal finance management apps aggregating balances across multiple banks.
- Business expense tracking and reconciliation tools.

## Access conditions {#conditions}

- The [Swan user](/topics/users/) must authenticate and grant explicit consent.
- The user must have an active [account membership](/topics/accounts/memberships/) with the `canViewAccount` permission.
- Consent must be valid, meaning it has not expired or been revoked.
- The connection must be active, meaning Salt Edge's daily refresh is working.

## Transaction history {#transaction-history}

- By default, Swan returns all transactions since account creation, including cards and transfers.
- TPPs commonly filter results to the last 3 months on their end.
- Some TPPs expose an endpoint that lets users retrieve transaction history beyond 3 months.

## Account scope {#account-scope}

By default, Swan returns all accounts the user has access to.

:::note All or nothing

> The consent flow does not currently allow users to select a subset of accounts; it is all or nothing.

Reviewer: ```suggestion
The consent flow requires users to give access to all accounts at once. Users cannot choose specific accounts to include.
```

" All or nothing" isn't a great heading for this; recommend making it more user-focused.

> This is a known limitation of the current consent flow.

Reviewer: Should we be calling this out as a limitation? Isn't just stating the behaviour enough?

:::

## No accounts visible after authentication {#troubleshooting-no-accounts}

This most often happens when a user authenticates with their **personal phone number**, but their Swan account is tied to their **professional phone number**.
The phone number is Swan's unique [user](/topics/users/) identifier, so authentication succeeds but returns no accounts.

**Solutions**:

> - **Option 1, try with the correct phone number**: authenticate again using the professional phone number linked to the Swan account.

Reviewer: ```suggestion
- **Option 1, try with the correct phone number**: authenticate again using the phone number linked to the Swan account.
```

| `amount.value` | `amount` | Negative string for debit, positive for credit. |
| `amount.currency` | `currency` | Currency code, such as `EUR` or `USD`. |
| `statusInfo.bookingDate` | `booking_date` | Optional. Date in `YYYY-MM-DD` format. |
| `statusInfo.status` | `status` | Only `Booked` and `Pending` transactions are exposed. `Rejected` transactions are not served through AIS. |
| `statusInfo.valueDate` when `Booked`, otherwise `updatedAt` | `value_date` | Date in `YYYY-MM-DD` format. When the transaction is not booked, this refers to the `updatedAt` date. |
| `label` | `remittance_information.unstructured` | Transaction label or description. Swan defaults to something like `Transfer to/from X` if the input was empty when initiating a SEPA Credit Transfer. |
| `reference` | `remittance_information.structured` | Structured payment reference. |
| `paymentId` | `extra.additional_information` | Not the same as the transaction ID. A payment can have multiple transactions, for example for fees. |
| For SCTs: `creditor.name`. Else if user is on the credit side: `account.holder.info.name`. Else: `counterparty`. | `creditor_details.name` | Creditor or recipient name. |

> | `creditor.IBAN` | `creditor_details.account.iban` | Optional. Always served when the transaction is a SEPA Credit Transfer or the end user is on the credit side. |

Reviewer: ```suggestion
| `creditor.IBAN` | `creditor_details.account.iban` | Optional. Always served when the transaction is a SEPA Credit Transfer or the user is on the credit side. |
```

Currently use 'user' and 'end user' interchangeably throughout the content. Update to user across everything.
